lib/plugins/deploy.js

```diff
@@ -36,6 +36,10 @@ class Deploy {
           throw new ServerlessError(errorMessage, 'INVALID_PROVIDER');
         }
         if (this.options.function) {
```

Comment:
I've consulted the implementation proposal in the ticket and I believe the suggestion was to implement it as a part of newly introduced `initialize` hook. Was there a reason to not follow the proposal here? The general approach is to emit deprecations as a part of `initialize` hooks